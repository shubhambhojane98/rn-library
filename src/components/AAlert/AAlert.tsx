import React from 'react';
import {StyleSheet, View, Modal, Pressable} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import {defaultScale} from '../../utils/Common';
import ATypography from '../../components/ATypography/ATypography';
import AButton from '../../components/AButton/AButton';
import {TypographyVariant} from '../ATypography/ATypographyEnum';
import {withTheme, useTheme} from '../../core/theming';
import type {Theme} from '../../utils/types';
interface Props {
  visible: boolean;
  title: string;
  onPressYes: () => void;
  onDismiss: () => void;
  subTitle?: string;
  description?: string;
  titleColor?: string;
  animationType?: 'none' | 'fade' | 'slide';
  backgroundColor?: string;
  borderRadius?: number;
  yesButtonColor?: string;
  noButtonColor?: string;
  theme: Theme;
}

const AAlert: React.FC<Props> = ({
  backgroundColor,
  visible,
  borderRadius,
  animationType,
  title,
  titleColor,
  subTitle,
  description,
  onPressYes,
  onDismiss,
  yesButtonColor,
  noButtonColor,
}) => {
  const {colors} = useTheme();

  const stylesWithProp = styles({colors});
  return (
    <Modal
      transparent
      animationType={animationType ? animationType : 'fade'}
      visible={visible}>
      <Pressable style={stylesWithProp.modalOverlay} onPress={onDismiss} />
      <View
        style={{
          ...stylesWithProp.modalView,
          backgroundColor: backgroundColor || colors.background,
          borderRadius,
        }}>
        <ATypography
          fontSize={moderateScale(18, defaultScale)}
          variant={TypographyVariant.PRIMARY_BOLD}
          color={titleColor || colors.textColor}
          style={stylesWithProp.typography}>
          {title}
        </ATypography>
        {subTitle ? (
          <ATypography
            fontSize={moderateScale(16, defaultScale)}
            variant={TypographyVariant.PRIMARY_SEMI_BOLD}
            color={colors.textColor}
            style={stylesWithProp.typography}>
            {subTitle}
          </ATypography>
        ) : null}
        {description ? (
          <ATypography
            fontSize={moderateScale(14, defaultScale)}
            variant={TypographyVariant.SECONDARY_DEMI}
            color={colors.textColor}
            style={stylesWithProp.typography}>
            {description}
          </ATypography>
        ) : null}

        <View style={stylesWithProp.Button}>
          <AButton
            onPress={onPressYes}
            title={'Yes'}
            backgroundColor={yesButtonColor || colors.primary}
            borderRadius={moderateScale(49, defaultScale)}
            margin={moderateScale(10, defaultScale)}
            width={moderateScale(120, defaultScale)}
          />
          <AButton
            title={'No'}
            onPress={onDismiss}
            backgroundColor={noButtonColor || colors.greyOpac10}
            borderRadius={moderateScale(49, defaultScale)}
            margin={moderateScale(10, defaultScale)}
            width={moderateScale(120, defaultScale)}
          />
        </View>
      </View>
    </Modal>
  );
};
const styles = (props: {colors: any}) =>
  StyleSheet.create({
    modalView: {
      position: 'absolute',
      alignItems: 'center',
      borderRadius: moderateScale(16, defaultScale),
      top: moderateScale(56, defaultScale),
      left: moderateScale(25, defaultScale),
      right: moderateScale(25, defaultScale),
    },
    modalOverlay: {
      position: 'absolute',
      top: moderateScale(0, defaultScale),
      right: moderateScale(0, defaultScale),
      bottom: moderateScale(0, defaultScale),
      left: moderateScale(0, defaultScale),
      backgroundColor: props.colors.backgroundLayout,
    },
    typography: {
      textAlign: 'center',
      paddingTop: moderateScale(25, defaultScale),
      paddingHorizontal: moderateScale(25, defaultScale),
    },
    Button: {
      flex: 1,
      flexDirection: 'row',
      paddingVertical: moderateScale(20, defaultScale),
    },
  });
export default withTheme(AAlert);
