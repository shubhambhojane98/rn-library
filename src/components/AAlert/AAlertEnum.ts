export enum AlertButtons {
  Yes = 'Yes',
  No = 'No',
}
