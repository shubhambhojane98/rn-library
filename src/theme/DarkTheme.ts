import type {Theme} from '../utils/types';

let DarkTheme: Theme;

DarkTheme = {
  dark: true,
  colors: {
    primary: '#BB86FC',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#262833',
    error: '#CF6679',
    textColor: '#FFF',
    shadowColor: '#D7D7D7',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
  },
};

export default DarkTheme;
