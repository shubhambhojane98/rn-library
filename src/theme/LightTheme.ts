import type {Theme} from '../utils/types';

let LightTheme: Theme;

LightTheme = {
  dark: false,
  colors: {
    primary: '#6200EE',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#FFF',
    error: '#B00020',
    textColor: '#000',
    shadowColor: '#222',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
    backgroundLayout: 'rgba(0, 0, 0, 0.8)',
    transparent: 'transparent',
    linkColor: '#0000EE',
    greyOpac50: 'rgba(53, 53, 53, 0.5)',
    greyOpac10: 'rgba(53, 53, 53, 0.1)',
  },
};

export default LightTheme;
