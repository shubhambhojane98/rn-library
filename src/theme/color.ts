import {Appearance} from 'react-native';

interface colors {
  yellow: string;
  black: string;
  red: string;
  violet: string;
  taupe: string;
  beige: string;
  grey: string;
  lightgrey: string;
  blue: string;
  darkGrey: string;
  white: string;
  transportation: string;
  logistic: string;
  environment: string;
  sports: string;
  healthcare: string;
  background: string;
  textColor: string;
  placeholderTextColor: string;
  switchBackgroundColor: string;
  paginationDotColor: string;
  paginationDotBgColor: string;
  blackOpac80: string;
  blackOpac60: string;
  blackOpac40: string;
  blackOpac20: string;
  greyOpac80: string;
  greyOpac60: string;
  greyOpac50: string;
  greyOpac40: string;
  greyOpac20: string;
  greyOpac10: string;
  switchActiveBgColor: string;
  switchInActiveBgColor: string;
  switchKnobColor: string;
}

const primaryColors = {
  yellow: '#FAD000',
  black: '#231F20',
  red: '#A81E23',
  violet: '#533467',
  taupe: '#695958',
  beige: '#C3B59F',
  grey: '#353535',
  white: '#fff',
  lightgrey: '#D7D7D7',
  blue: '#007BC1',
  darkGrey: 'rgba(0, 0, 0, 0.2)',
};

const opacColors = {
  blackOpac80: 'rgba(0, 0, 0, 0.8)',
  blackOpac60: 'rgba(0, 0, 0, 0.6)',
  blackOpac40: 'rgba(0, 0, 0, 0.4)',
  blackOpac20: 'rgba(0, 0, 0, 0.2)',
  greyOpac80: 'rgba(53, 53, 53, 0.8)',
  greyOpac60: 'rgba(53, 53, 53, 0.6)',
  greyOpac50: 'rgba(53, 53, 53, 0.5)',
  greyOpac40: 'rgba(53, 53, 53, 0.4)',
  greyOpac20: 'rgba(53, 53, 53, 0.2)',
  greyOpac10: 'rgba(53, 53, 53, 0.1)',
};

const categoryColors = {
  transportation: '#007BC1',
  logistic: '#EF4065',
  environment: '#009A46',
  sports: '#F26524',
  healthcare: '#00ABBE',
};

let themeColors: colors;

const colorScheme = Appearance.getColorScheme();
if (colorScheme === 'dark') {
  // Use dark color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.black,
    textColor: primaryColors.white,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.white,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
} else {
  // Use light color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.yellow,
    textColor: primaryColors.black,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.lightgrey,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
}

export default themeColors;
