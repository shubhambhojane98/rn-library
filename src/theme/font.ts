import { Platform, PlatformOSType } from 'react-native';
import { Fonts } from '../utils/types';


export default function configureFonts(config?: {
    [platform in PlatformOSType | 'default']?: Fonts;
  }): Fonts {
    const fonts = Platform.select({ ...config }) as Fonts;
    return fonts;
  }