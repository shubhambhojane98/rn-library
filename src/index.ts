import {defaultScale} from './utils/Common';
import {consoleLog, consoleWarn, consoleInfo} from './utils/Log';
import {CheckboxStatus} from './components/ACheckbox/ACheckboxEnum';
import {Color} from './theme';
export {default as AButton} from './components/AButton/AButton';
export {default as ATypography} from './components/ATypography/ATypography';
export {default as AAvatar} from './components/AAvatar/AAvatar';
export {default as ABadge} from './components/ABadge/ABadge';
export {default as AImage} from './components/AImage/AImage';
export {default as ARadioButton} from './components/ARadioButton/ARadioButton';
export {default as ARadioButtonGroup} from './components/ARadioButtonGroup/ARadioButtonGroup';
export {default as ATextInput} from './components/ATextInput/ATextInput';
export {default as ASwitch} from './components/ASwitch/ASwitch';
export {default as AVirtualizedList} from './components/AVirtualizedList/AVirtualizedList';
export {default as AModal} from './components/AModal/AModal';
export {default as ASwitchSelector} from './components/ASwitchSelector/ASwitchSelector';
export {default as AStarRating} from './components/AStarRating/AStarRating';
export {default as ACheckbox} from './components/ACheckbox/ACheckbox';
export {default as AToast} from './components/AToast/AToast';
export {default as ACard} from './components/ACard/ACard';
export {consoleLog, consoleWarn, consoleInfo};
export {Color};
export {CheckboxStatus};
