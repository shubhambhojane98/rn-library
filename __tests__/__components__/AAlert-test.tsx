import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AAlert from '../../src/components/AAlert/AAlert';
import {Color} from '../../src/theme';

it('renders correctly with default props', () => {
  const tree = renderer
    .create(
      <AAlert
        visible={true}
        title={'Title'}
        onPressYes={jest.fn()}
        onDismiss={jest.fn()}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with   Alert with backgroundColor, visible, borderRadius, animationType, title, titleColor, subTitle, description, onPressYes, onDismiss,', () => {
  const tree = renderer
    .create(
      <AAlert
        visible={true}
        title={'Title'}
        titleColor={'red'}
        onPressYes={jest.fn()}
        subTitle={'Sub_Title'}
        description={'Description'}
        onDismiss={jest.fn()}
        backgroundColor={Color.healthcare}
        borderRadius={8}
        animationType={'fade'}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with visible false', () => {
  const tree = renderer
    .create(
      <AAlert
        visible={false}
        title={'Title'}
        onPressYes={jest.fn()}
        onDismiss={jest.fn()}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
