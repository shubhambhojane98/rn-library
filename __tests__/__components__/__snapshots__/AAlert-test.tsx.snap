// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with   Alert with backgroundColor, visible, borderRadius, animationType, title, titleColor, subTitle, description, onPressYes, onDismiss, 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.8)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00ABBE",
        "borderRadius": 8,
        "left": 30.714285714285715,
        "position": "absolute",
        "right": 30.714285714285715,
        "top": 68.8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Bold",
            "fontSize": 27.168979591836734,
            "fontWeight": "bold",
          },
          Object {
            "color": "red",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "paddingHorizontal": 30.714285714285715,
            "paddingTop": 30.714285714285715,
            "textAlign": "center",
          },
        ]
      }
      theme={
        Object {
          "colors": Object {
            "background": "#FFF",
            "backgroundLayout": "rgba(0, 0, 0, 0.8)",
            "black": "#000",
            "error": "#B00020",
            "greyOpac10": "rgba(53, 53, 53, 0.1)",
            "greyOpac50": "rgba(53, 53, 53, 0.5)",
            "lightgrey": "#D7D7D7",
            "linkColor": "#0000EE",
            "primary": "#FAD000",
            "primaryVariant": "#3700B3",
            "secondary": "#03DAC6",
            "secondaryVariant": "#018786",
            "shadowColor": "#222",
            "textColor": "#000",
            "transparent": "transparent",
            "white": "#FFF",
          },
          "dark": false,
        }
      }
    >
      Title
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-SemiBold",
            "fontSize": 24.150204081632655,
            "fontWeight": "600",
          },
          Object {
            "color": "#000",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "paddingHorizontal": 30.714285714285715,
            "paddingTop": 30.714285714285715,
            "textAlign": "center",
          },
        ]
      }
      theme={
        Object {
          "colors": Object {
            "background": "#FFF",
            "backgroundLayout": "rgba(0, 0, 0, 0.8)",
            "black": "#000",
            "error": "#B00020",
            "greyOpac10": "rgba(53, 53, 53, 0.1)",
            "greyOpac50": "rgba(53, 53, 53, 0.5)",
            "lightgrey": "#D7D7D7",
            "linkColor": "#0000EE",
            "primary": "#FAD000",
            "primaryVariant": "#3700B3",
            "secondary": "#03DAC6",
            "secondaryVariant": "#018786",
            "shadowColor": "#222",
            "textColor": "#000",
            "transparent": "transparent",
            "white": "#FFF",
          },
          "dark": false,
        }
      }
    >
      Sub_Title
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "AvenirNextLTPro-Demi",
            "fontSize": 21.13142857142857,
            "fontWeight": "600",
          },
          Object {
            "color": "#000",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "paddingHorizontal": 30.714285714285715,
            "paddingTop": 30.714285714285715,
            "textAlign": "center",
          },
        ]
      }
      theme={
        Object {
          "colors": Object {
            "background": "#FFF",
            "backgroundLayout": "rgba(0, 0, 0, 0.8)",
            "black": "#000",
            "error": "#B00020",
            "greyOpac10": "rgba(53, 53, 53, 0.1)",
            "greyOpac50": "rgba(53, 53, 53, 0.5)",
            "lightgrey": "#D7D7D7",
            "linkColor": "#0000EE",
            "primary": "#FAD000",
            "primaryVariant": "#3700B3",
            "secondary": "#03DAC6",
            "secondaryVariant": "#018786",
            "shadowColor": "#222",
            "textColor": "#000",
            "transparent": "transparent",
            "white": "#FFF",
          },
          "dark": false,
        }
      }
    >
      Description
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 24.57142857142857,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FAD000",
            "borderColor": undefined,
            "borderRadius": 60.2,
            "borderWidth": undefined,
            "flexDirection": "row",
            "height": 61.42857142857143,
            "justifyContent": "center",
            "margin": 15.093877551020407,
            "opacity": 1,
            "textAlign": "center",
            "width": 181.1265306122449,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Mulish-Bold",
                "fontSize": 21.13142857142857,
                "fontWeight": "bold",
              },
              Object {
                "color": "#000",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "alignItems": "center",
                "fontFamily": undefined,
                "fontSize": 18,
                "fontWeight": undefined,
                "padding": 12.285714285714285,
                "textAlign": "center",
                "textDecorationLine": undefined,
              },
            ]
          }
          theme={
            Object {
              "colors": Object {
                "background": "#FFF",
                "backgroundLayout": "rgba(0, 0, 0, 0.8)",
                "black": "#000",
                "error": "#B00020",
                "greyOpac10": "rgba(53, 53, 53, 0.1)",
                "greyOpac50": "rgba(53, 53, 53, 0.5)",
                "lightgrey": "#D7D7D7",
                "linkColor": "#0000EE",
                "primary": "#FAD000",
                "primaryVariant": "#3700B3",
                "secondary": "#03DAC6",
                "secondaryVariant": "#018786",
                "shadowColor": "#222",
                "textColor": "#000",
                "transparent": "transparent",
                "white": "#FFF",
              },
              "dark": false,
            }
          }
        >
          Yes
        </Text>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FAD000",
            "borderColor": undefined,
            "borderRadius": 60.2,
            "borderWidth": undefined,
            "flexDirection": "row",
            "height": 61.42857142857143,
            "justifyContent": "center",
            "margin": 15.093877551020407,
            "opacity": 1,
            "textAlign": "center",
            "width": 181.1265306122449,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Mulish-Bold",
                "fontSize": 21.13142857142857,
                "fontWeight": "bold",
              },
              Object {
                "color": "#000",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "alignItems": "center",
                "fontFamily": undefined,
                "fontSize": 18,
                "fontWeight": undefined,
                "padding": 12.285714285714285,
                "textAlign": "center",
                "textDecorationLine": undefined,
              },
            ]
          }
          theme={
            Object {
              "colors": Object {
                "background": "#FFF",
                "backgroundLayout": "rgba(0, 0, 0, 0.8)",
                "black": "#000",
                "error": "#B00020",
                "greyOpac10": "rgba(53, 53, 53, 0.1)",
                "greyOpac50": "rgba(53, 53, 53, 0.5)",
                "lightgrey": "#D7D7D7",
                "linkColor": "#0000EE",
                "primary": "#FAD000",
                "primaryVariant": "#3700B3",
                "secondary": "#03DAC6",
                "secondaryVariant": "#018786",
                "shadowColor": "#222",
                "textColor": "#000",
                "transparent": "transparent",
                "white": "#FFF",
              },
              "dark": false,
            }
          }
        >
          No
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`renders correctly with default props 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.8)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": undefined,
        "left": 30.714285714285715,
        "position": "absolute",
        "right": 30.714285714285715,
        "top": 68.8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Bold",
            "fontSize": 27.168979591836734,
            "fontWeight": "bold",
          },
          Object {
            "color": "#000",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "paddingHorizontal": 30.714285714285715,
            "paddingTop": 30.714285714285715,
            "textAlign": "center",
          },
        ]
      }
      theme={
        Object {
          "colors": Object {
            "background": "#FFF",
            "backgroundLayout": "rgba(0, 0, 0, 0.8)",
            "black": "#000",
            "error": "#B00020",
            "greyOpac10": "rgba(53, 53, 53, 0.1)",
            "greyOpac50": "rgba(53, 53, 53, 0.5)",
            "lightgrey": "#D7D7D7",
            "linkColor": "#0000EE",
            "primary": "#FAD000",
            "primaryVariant": "#3700B3",
            "secondary": "#03DAC6",
            "secondaryVariant": "#018786",
            "shadowColor": "#222",
            "textColor": "#000",
            "transparent": "transparent",
            "white": "#FFF",
          },
          "dark": false,
        }
      }
    >
      Title
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 24.57142857142857,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FAD000",
            "borderColor": undefined,
            "borderRadius": 60.2,
            "borderWidth": undefined,
            "flexDirection": "row",
            "height": 61.42857142857143,
            "justifyContent": "center",
            "margin": 15.093877551020407,
            "opacity": 1,
            "textAlign": "center",
            "width": 181.1265306122449,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Mulish-Bold",
                "fontSize": 21.13142857142857,
                "fontWeight": "bold",
              },
              Object {
                "color": "#000",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "alignItems": "center",
                "fontFamily": undefined,
                "fontSize": 18,
                "fontWeight": undefined,
                "padding": 12.285714285714285,
                "textAlign": "center",
                "textDecorationLine": undefined,
              },
            ]
          }
          theme={
            Object {
              "colors": Object {
                "background": "#FFF",
                "backgroundLayout": "rgba(0, 0, 0, 0.8)",
                "black": "#000",
                "error": "#B00020",
                "greyOpac10": "rgba(53, 53, 53, 0.1)",
                "greyOpac50": "rgba(53, 53, 53, 0.5)",
                "lightgrey": "#D7D7D7",
                "linkColor": "#0000EE",
                "primary": "#FAD000",
                "primaryVariant": "#3700B3",
                "secondary": "#03DAC6",
                "secondaryVariant": "#018786",
                "shadowColor": "#222",
                "textColor": "#000",
                "transparent": "transparent",
                "white": "#FFF",
              },
              "dark": false,
            }
          }
        >
          Yes
        </Text>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FAD000",
            "borderColor": undefined,
            "borderRadius": 60.2,
            "borderWidth": undefined,
            "flexDirection": "row",
            "height": 61.42857142857143,
            "justifyContent": "center",
            "margin": 15.093877551020407,
            "opacity": 1,
            "textAlign": "center",
            "width": 181.1265306122449,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Mulish-Bold",
                "fontSize": 21.13142857142857,
                "fontWeight": "bold",
              },
              Object {
                "color": "#000",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "alignItems": "center",
                "fontFamily": undefined,
                "fontSize": 18,
                "fontWeight": undefined,
                "padding": 12.285714285714285,
                "textAlign": "center",
                "textDecorationLine": undefined,
              },
            ]
          }
          theme={
            Object {
              "colors": Object {
                "background": "#FFF",
                "backgroundLayout": "rgba(0, 0, 0, 0.8)",
                "black": "#000",
                "error": "#B00020",
                "greyOpac10": "rgba(53, 53, 53, 0.1)",
                "greyOpac50": "rgba(53, 53, 53, 0.5)",
                "lightgrey": "#D7D7D7",
                "linkColor": "#0000EE",
                "primary": "#FAD000",
                "primaryVariant": "#3700B3",
                "secondary": "#03DAC6",
                "secondaryVariant": "#018786",
                "shadowColor": "#222",
                "textColor": "#000",
                "transparent": "transparent",
                "white": "#FFF",
              },
              "dark": false,
            }
          }
        >
          No
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`renders correctly with visible false 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
/>
`;
